_posts/2021/2021-05-24-opt-out-of-google-floc-tracking-in-rails.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Opt out of Google’s FLoC User Tracking in Rails"
+description: "FLoC Off"
+layout: article
+category: ruby
+image:
+  base: "2021/opt-out-of-google-floc-tracking"
+  alt: "V for Vendetta-style Guy Fawes masks"
+  source: "https://unsplash.com/photos/RlLP5fUh2m0"
+  credit: " Ahmed Zayan"
+---
+
+Google has recently built user-tracking directly into Chrome. This new tracking technology is named [Federated Learning of Cohorts](https://blog.google/products/chrome/privacy-sustainability-and-the-importance-of-and/) (shortened to <abbr title="Federated Learning of Cohorts">FLoC</abbr>).
+
+
+## Instead of...
+
+...doing nothing.
+
+
+## Add...
+
+...a custom response header to opt your application, and thus your users, out of this user-tracking.
+
+### `config/appliction.rb`
+
+```ruby
+# ...
+module YourAppName
+  class Application < Rails::Application
+    # ...
+    config.action_dispatch.default_headers["Permissions-Policy"] = "interest-cohort=()"
+    # ...
+  end
+end
+```
+
+
+## Why?
+
+The [<abbr title="Electronic Frontier Foundation">EFF</abbr>](https://www.eff.org) has a detailed breakdown of [why FLoC is a bad idea](https://amifloced.org) as well as checking if you’re included in their initial roll out.
+
+In short it's not just bad because advertisers can track you around the internet (still creepy after all these years!) but also because it further entrenches Google’s advertising monopoly by leveraging their dominance in the browser market.
```

Comment:
```suggestion
In short, it's not just bad because advertisers can track you around the Internet (still creepy after all these years!) but also because Google are leveraging their dominance in the browser market to further entrench their advertising monopoly.
```